src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -0,0 +1,329 @@
+
+            this.addRegion((incrementalRegionID++).toString(), commit, tags);
+
+            // RegionData not serializable so need to extract data
+            const scaledRegionData = this.scaleRegionToSourceSize(commit);
+            const newRegion = {
+                id: incrementalRegionID.toString(),
+                type: RegionType.Rectangle,
+                tags,
+                points: [new Point2D(scaledRegionData.x, scaledRegionData.y),
+                        new Point2D(scaledRegionData.x + scaledRegionData.width,
+                                    scaledRegionData.y + scaledRegionData.height)],
+            };
+
+            const currentAssetMetadata = this.props.selectedAsset;
+            currentAssetMetadata.regions.push(newRegion);
+            if (currentAssetMetadata.regions.length) {
+                currentAssetMetadata.asset.state = AssetState.Tagged;
+            }
+
+            this.props.onAssetMetadataChanged(currentAssetMetadata);
+        };
+
+        this.editor.onRegionMove = (id, regionData) => {
+            const currentAssetMetadata = this.props.selectedAsset;
+            const movedRegionIndex = currentAssetMetadata.regions.findIndex((region) => region.id === id);
+            const movedRegion = currentAssetMetadata.regions[movedRegionIndex];
+            // @ts-ignore   in here until CanvasTools types get updated
+            const scaledRegionData = this.scaleRegionToSourceSize(regionData);
+            if (movedRegion) {
+                movedRegion.points = [new ct.Core.Point2D(scaledRegionData.x, scaledRegionData.y),
+                                    new ct.Core.Point2D(scaledRegionData.x + scaledRegionData.width,
+                                                        scaledRegionData.y + scaledRegionData.height)];
+
+            }
+            currentAssetMetadata.regions[movedRegionIndex] = movedRegion;
+            this.props.onAssetMetadataChanged(currentAssetMetadata);
+        };
+
+        this.editor.onRegionDelete = (id) => {
+            this.deleteRegionById(id);
+            const currentAssetMetadata = this.props.selectedAsset;
+            const deletedRegionIndex = this.props.selectedAsset.regions.findIndex((region) => region.id === id);
+            currentAssetMetadata.regions.splice(deletedRegionIndex, 1);
+            if (!currentAssetMetadata.regions.length) {
+                currentAssetMetadata.asset.state = AssetState.Visited;
+            }
+            this.props.onAssetMetadataChanged(currentAssetMetadata);
+        };
+
+        // Upload background image for selection
+        this.updateEditor();
+    }
+
+    public componentDidUpdate(prevProps) {
+        if (this.props.selectedAsset.asset.path !== prevProps.selectedAsset.asset.path) {
+            this.updateEditor();
+        }
+    }
+
+    public render() {
+        // const { loaded } = this.state;
+        // const { svgHost } = this.props;
+
+        return (
+            <div id="ctZone">
```

Comment:
Should this contain the CanvasTools component? Has that been reactified yet?

Reply:
So currently this is how canvas tools is initialized. It uses those id names to generate the editor.